flytepropeller/pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -625,34 +626,22 @@ func (in *NodeStatus) UpdatePhase(p NodePhase, occurredAt metav1.Time, reason st
 			in.LastAttemptStartedAt = &n
 		}
 	} else if IsPhaseTerminal(p) {
-		// If we are in terminal phase then we will clear out all our fields as they are not required anymore
-		// Only thing required is stopped at and lastupdatedat time
 		if in.StoppedAt == nil {
 			in.StoppedAt = &n
 		}
-		if in.StartedAt == nil {
-			in.StartedAt = &n
-		}
-		if in.LastAttemptStartedAt == nil {
-			in.LastAttemptStartedAt = &n
-		}
-	}
-	in.LastUpdatedAt = &n
-
-	// For cases in which the node is either Succeeded or Skipped we clear most fields from the status
-	// except for StoppedAt and Phase. StoppedAt is used to calculate transition latency between this node and
-	// any downstream nodes and Phase is required for propeller to continue to downstream nodes.
-	if p == NodePhaseSucceeded || p == NodePhaseSkipped {
+		// Clear most status related fields after reaching a terminal state. This keeps the CRD state small to avoid 
+		// etcd size limits. Importantly we keep Phase, StoppedAt and Error which will be needed further. 
+		// Errors will still be needed but it will be cleaned up when possible because they can be very large.  
```

Comment:
Is there a way we can gate this behind a flag? I know some users like having error messages and status' persist in the CR when nodes fail.

Reply:
There must be a way but I don't really know how. Do you have any suggestions? 

I think one way would be to add a `ClearStateOnTermination` attribute to `NodeStatus`. Then if we set it correctly when creating the node everything should work after that. The disadvantage of that though is we create a new parameter that needs to be stored in etcd. Personally I think it would be preferable to pass in an argument to `UpdatePhase`. It seems like this would require a change to a commonly used interface and I don't really know what the implications of that would be.

Reply:
I opened a draft upstream PR so its probably best to discuss there https://github.com/flyteorg/flyte/pull/4596